subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/AbstractPluginValidationIntegrationSpec.groovy

```diff
@@ -0,0 +1,766 @@
+        )
+    }
+
+    def "detects annotations on setter methods"() {
+        javaTaskSource << """
+            import org.gradle.api.*;
+            import org.gradle.api.tasks.*;
+            import java.io.File;
+
+            public class MyTask extends DefaultTask {
+                @Input
+                public void setWriteOnly(String value) {
+                }
+
+                public String getReadWrite() {
+                    return "read-write property";
+                }
+
+                @Input
+                public void setReadWrite(String value) {
+                }
+
+                @Nested
+                public Options getOptions() {
+                    return new Options();
+                }
+
+                public static class Options {
+                    @Input
+                    public void setWriteOnly(String value) {
+                    }
+
+                    @Input
+                    public String getReadWrite() {
+                        return "read-write property";
+                    }
+
+                    @Input
+                    public void setReadWrite(String value) {
+                    }
+                }
+
+                @TaskAction
+                public void doStuff() { }
+            }
+        """
+
+        expect:
+        assertValidationFailsWith(
+            "Type 'MyTask\$Options': setter method 'setReadWrite()' should not be annotated with: @Input": WARNING,
+            "Type 'MyTask\$Options': setter method 'setWriteOnly()' should not be annotated with: @Input": WARNING,
+            "Type 'MyTask': property 'readWrite' is not annotated with an input or output annotation.": WARNING,
+            "Type 'MyTask': setter method 'setReadWrite()' should not be annotated with: @Input": WARNING,
+            "Type 'MyTask': setter method 'setWriteOnly()' should not be annotated with: @Input": WARNING,
+        )
+    }
+
+    @Requires(TestPrecondition.JDK8_OR_LATER)
+    def "can validate task classes using external types"() {
+        taskBuildFile << """
+            ${jcenterRepository()}
+
+            dependencies {
+                implementation 'com.typesafe:config:1.3.2'
+            }
+        """
+
+        javaTaskSource << """
+            import org.gradle.api.*;
+            import org.gradle.api.tasks.*;
+            import java.io.File;
+            import com.typesafe.config.Config;
+
+            public class MyTask extends DefaultTask {
+                @Input
+                public long getGoodTime() {
+                    return 0;
+                }
+                
+                @Optional @Input
+                public Config getConfig() { return null; } 
+                
+                @TaskAction public void execute() {}
+            }
+        """
+
+        expect:
+        assertValidationSucceeds()
+    }
+
+    @Requires(TestPrecondition.JDK8_OR_LATER)
+    def "can validate task classes using types from other projects"() {
```

Comment:
Why do we need this requires? Gradle doesn't run on JDK < 8.

It also seems that this task only belongs to the ValidatePluginsIntegrationTest.